client/src/rr/dnssec/keypair.rs

```diff
@@ -456,51 +452,6 @@ impl KeyPair {
     }
 }
 
-impl PublicKey for KeyPair {
```

Comment:
huh, I had thought this would make things easier supporting multiple versions of keys, I'm fine with removing it if you find it not to do that.

Reply:
I think it's better to remove it. It's easy to extract the public key bytes from the key pair and construct a PublicKeyEnum from them, so this code isn't doing anything that can't be done otherwise. And, it requires extra work to maintain (which is how I stumbled across the idea for this PR).